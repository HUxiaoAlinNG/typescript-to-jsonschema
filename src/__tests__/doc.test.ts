import * as path from 'path';

import genTypeSchema from '../index';

const file = path.join(__dirname, `../../jest/doc.ts`);
genTypeSchema.genJsonDataFormFile(file);

const getSchema = (type: string) => {
  const schema = genTypeSchema.getJsonSchema(file, type);
  return schema;
};

test('单行注释', () => {
  expect(getSchema('doc_1')).toMatchSnapshot();
});

test('多行注释', () => {
  expect(getSchema('doc_2')).toMatchSnapshot();
});

test('单行注释默认值', () => {
  expect(getSchema('doc_3')).toMatchSnapshot();
});

test('多行注释默认值', () => {
  expect(getSchema('doc_4')).toMatchSnapshot();
});

test('单多行注释默认值', () => {
  expect(getSchema('doc_5')).toMatchSnapshot();
});
