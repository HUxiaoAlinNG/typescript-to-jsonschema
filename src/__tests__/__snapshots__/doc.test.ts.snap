// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`单多行注释默认值 1`] = `
Object {
  "additionalProperties": false,
  "properties": Object {
    "age": Object {
      "default": "25",
      "description": "年龄",
      "type": "number",
    },
    "name": Object {
      "default": "'zane'",
      "description": "姓名",
      "type": "string",
    },
  },
  "required": Array [
    "name",
    "age",
  ],
  "type": "object",
}
`;

exports[`单行注释 1`] = `
Object {
  "additionalProperties": false,
  "properties": Object {
    "name": Object {
      "description": "支持单行注释",
      "type": "string",
    },
  },
  "required": Array [
    "name",
  ],
  "type": "object",
}
`;

exports[`单行注释默认值 1`] = `
Object {
  "additionalProperties": false,
  "properties": Object {
    "name": Object {
      "default": "'zane'",
      "description": "姓名",
      "type": "string",
    },
  },
  "required": Array [
    "name",
  ],
  "type": "object",
}
`;

exports[`多行注释 1`] = `
Object {
  "additionalProperties": false,
  "properties": Object {
    "name": Object {
      "description": "支持多行注释",
      "type": "string",
    },
  },
  "required": Array [
    "name",
  ],
  "type": "object",
}
`;

exports[`多行注释默认值 1`] = `
Object {
  "additionalProperties": false,
  "properties": Object {
    "age": Object {
      "default": "25",
      "description": "年龄",
      "type": "number",
    },
  },
  "required": Array [
    "age",
  ],
  "type": "object",
}
`;
